tests/mikro-orm.module.test.ts

```diff
@@ -232,6 +233,29 @@ describe('MikroORM Module', () => {
       await orm.close();
     });
 
+    it('forFeature should return repository for schema', async () => {
+      const module = await Test.createTestingModule({
+        imports: [
+          MikroOrmModule.forRoot({
+            ...testOptions,
+            entities: undefined,
+            autoLoadEntities: true,
+          }),
+          MikroOrmModule.forFeature([baseEntitySchema, bazEntitySchema]),
```

Comment:
ok, now i get whats happening. this is rather misusage, you are not supposed to be putting base entities into forFeature calls. you should be doing this:

```suggestion
          MikroOrmModule.forRoot({
            ...testOptions,
            entities: [baseEntitySchema], // put only base entities here
            autoLoadEntities: true,
          }),
          MikroOrmModule.forFeature([bazEntitySchema]),
```

i get that this is rather confusing and i dont mind ignoring base classes instead of failing, but it should be consistent - It needs to work the same for base entities defined with decorators too

Reply:
note that with decorator based definitions, you dont need to explicitly discover the base entities, it works automatically, so this is not a problem there

Reply:
Hey there, Martin! Thanks for the insights, I'll check this out later today carefully and devise a solution!

Reply:
FYI https://github.com/mikro-orm/mikro-orm/commit/cfdfa7eb3eaaff330f77eb92093b0fe4077568cb?w=1